Sources/Commands/SwiftTool.swift

```diff
@@ -245,6 +250,37 @@ extension SwiftCommand {
     public var toolWorkspaceConfiguration: ToolWorkspaceConfiguration {
         return .init()
     }
+
+    public func buildSystemProvider(_ swiftTool: SwiftTool) throws -> BuildSystemProvider {
+        return .init(providers: [
+            .native: { (explicitProduct: String?, cacheBuildManifest: Bool, customBuildParameters: BuildParameters?, customPackageGraphLoader: (() throws -> PackageGraph)?, customOutputStream: OutputByteStream?, customLogLevel: Diagnostic.Severity?, customObservabilityScope: ObservabilityScope?) throws -> BuildSystem in
+                let testEntryPointPath = customBuildParameters?.testProductStyle.explicitlySpecifiedEntryPointPath
+                let graphLoader = { try swiftTool.loadPackageGraph(explicitProduct: explicitProduct, testEntryPointPath: testEntryPointPath) }
+                return try BuildOperation(
+                buildParameters: customBuildParameters ?? swiftTool.buildParameters(),
+                cacheBuildManifest: cacheBuildManifest && swiftTool.canUseCachedBuildManifest(),
+                packageGraphLoader: customPackageGraphLoader ?? graphLoader,
+                additionalFileRules: FileRuleDescription.swiftpmFileTypes,
+                pluginScriptRunner: swiftTool.getPluginScriptRunner(),
+                pluginWorkDirectory: try swiftTool.getActiveWorkspace().location.pluginWorkingDirectory,
+                outputStream: customOutputStream ?? swiftTool.outputStream,
+                logLevel: customLogLevel ?? swiftTool.logLevel,
+                fileSystem: swiftTool.fileSystem,
+                observabilityScope: customObservabilityScope ?? swiftTool.observabilityScope) },
+            .xcode: { (explicitProduct: String?, cacheBuildManifest: Bool, customBuildParameters: BuildParameters?, customPackageGraphLoader: (() throws -> PackageGraph)?, customOutputStream: OutputByteStream?, customLogLevel: Diagnostic.Severity?, customObservabilityScope: ObservabilityScope?) throws -> BuildSystem in
+                let graphLoader = { try swiftTool.loadPackageGraph(explicitProduct: explicitProduct) }
+                // FIXME: Implement the custom build command provider also.
+                return try XcodeBuildSystem(
+                    buildParameters: customBuildParameters ?? swiftTool.buildParameters(),
+                    packageGraphLoader: customPackageGraphLoader ?? graphLoader,
+                    outputStream: customOutputStream ?? swiftTool.outputStream,
+                    logLevel: customLogLevel ?? swiftTool.logLevel,
+                    fileSystem: swiftTool.fileSystem,
+                    observabilityScope: customObservabilityScope ?? swiftTool.observabilityScope
+                )
+            }
+        ])
```

Comment:
there is a but of repetition here between the two cases that can be made DRYer

Reply:
Yah, that's intentional since the `.xcode` case will move out of this type in one of the next PRs.